### DISCUSSION TITLE
Roadmap & Vision

### DISCUSSION THREAD

**Discussion by Vadorequest** [nocodb/nocodb#194] [Q&A]
Thank you for releasing this open-source "alternative of Airtable".

I wonder what are your plans and vision for this tool. I saw you plan to release a hosted (managed) version.

OSS is great, but pricing/costs are often real issues, even for the best initiatives. There are countless OSS products that have fallen short finding funding to keep improving the product. It takes time and effort, and often, money.

I understand this project is powered by the **XgeneCloud** company. I couldn't find much on the company itself besides [the GitHub](https://github.com/xgenecloud). I'd love to know a bit more about the company's vision! Why did you decide to make NocoDB open source? What are your motivations? Do you use it internally? Do you hope to make money through the hosted version?

It's really great to see an alternative for Airtable, and the recent open-source announcement [seems to have many people hyped](http://www.gitstarhistory.com/#nocodb/nocodb)! (I am!)

![image](https://user-images.githubusercontent.com/3807458/120083261-bdc8a400-c0c7-11eb-859d-ba855a4207b8.png)

I bet the growing community would love to hear more about your grand plan. 😉

**Comment by o1lab** [comment]
Quite a few questions here 😃  

> Vision

# 🎯 Why are we building this ?
Most internet businesses equip themselves with either spreadsheet or a database to solve their business needs. Spreadsheets are used by a Billion+ humans collaboratively every single day. However, we are way off working at similar speeds on databases which are way more powerful tools when it comes to computing. Attempts to solve this with SaaS offerings has meant horrible access controls, vendor lockin, data lockin, abrupt price changes & most importantly a glass ceiling of what's possible in future.

# ❤ Our Mission :
Our mission is to provide the most powerful no-code interface for databases which is open source to every single internet business in the world. This would not only democratise access to a powerful computing tool but also bring forth a billion+ people who will have radical tinkering-and-building abilities on internet.

> Why did you decide to make NocoDB open source? What are your motivations?

# 🔦  Also how this all began ..
We had open sourced two API solutions before NocoDB :
⚡ A no-code REST APIs generator for any MySQL DB. ~200,000 Docker pulls. This was a hobby project & had no GUI. 📈
⚡ A low-code REST-GraphQL APIs generator for any database with GUI. Used by 100s of companies. Including fortune 500s & publicly trading companies. 📈

The thing that surprised us the most was that even non-developers started using our API products & rooting for us. 🚀 🚀  Whilst everybody loved instant API access to databases, it was slow-and-painful for them to build UI and collaborate with their teams. 💔 💔 ... This made us to radically recombine the best bits of our 2 API products then transform them into something better.


> Do you hope to make money through the hosted version?

Yes the idea is to build a hosting solution around this. This is so that our users can take advantage of a seamless setup. I believe the likes of AWS or not as seamless you like them to be. Having said that, we are in early days and will have to perfect the initial product from users feedback.

> Recent open-source announcement seems to have many people hyped! (I am!

We are overwhelmed by responses as well.

**Reply by Vadorequest** [reply]
Thanks for the reply. 

How are you financing this effort? I saw (on Discord) you're currently a team of 2, do you plan on recruiting?

What's the most difficult thing coming ahead at this time? How can the community help?

Do you have a raw idea about how will the hosted solution work, and how it will be priced? The idea isn't to commit to anything but simply get a glance at what you have in mind at this time. 😉 

I certainly hope you succeed! 💪
